db/db_test.cc

```diff
@@ -6118,9 +6134,14 @@ TEST_F(DBTest, EncodeDecompressedBlockSizeTest) {
         // compressible string
         ASSERT_OK(Put(Key(i), rnd.RandomString(128) + std::string(128, 'a')));
       }
+      ASSERT_OK(Flush());
 
       table_options.format_version = first_table_version == 1 ? 2 : 1;
       options.table_factory.reset(NewBlockBasedTableFactory(table_options));
+      // format_version (for writing) is sanitized to minimum supported
```

Comment:
> .. (for writing)... 

Nice clarification. Looking into the code now I know TableOptions::format_version matters in write table while  TableOptions::format_version + rocksdb version matters in read table

Reply:
Adding some comments to clarify. BBTO is not very clear on this.